### ISSUE TITLE
REQUEST: Add YAML support for nselib

### ISSUE THREAD

**Issue by xbol0** [nmap/nmap#2537]
YAML is a common configure format which being widely used, many tools used it as main configuring file or template, such as Nuclei, ffuf and ChopChop.

Support YAML in nselib, we can reuse these tool's template to make integration better.

**Comment by dmiller-nmap** [comment]
Thanks for the suggestion. Can you explain how you envision YAML being used within NSE? What kind of NSE script would benefit from this support?

**Comment by xbol0** [comment]
Yes, YAML is a format gradually popular, because of  human-friendly, which similar to JSON. Many applications use it as mainly template, like Nuclei. [Nuclei-templates](https://github.com/projectdiscovery/nuclei-templates) have many templates for vulnerabilities, a lot of guys contribute to it, we can reuse it for scan deeper. But currently we have to convert to JSON, it depends another service to process.

In the other fact, we can even output custom result as YAML, other applications can use it for next step.
